### PR TITLE
Add support for retrieving secrets from kubernetes

### PR DESCRIPTION
**Pull request by rhmdnd**
We want the ability to deploy compser on a kubernetes cluster, which
also means secrets (e.g., database passwords) would be held as
kubernetes secrets.

Let's add a method to config handling that uses kubernetes clients to
fetch secrets. We'll use this in subsequent patches that use kustomize
to deploy compserv and a database on kubernetes.

### CODE REVIEW

File: pkg/config/secrets.go [hunk 1]
```diff
@@ -1,16 +1,41 @@
...
+	"k8s.io/client-go/rest"
 )
 
+func getSecretFromKubernetes(v *viper.Viper) string {
+	c, err := rest.InClusterConfig()
+	if err != nil {
+		log.Fatalf("Failed to build Client for Kubernetes cluster: %s", err)
+	}
+
+	cs, err := kubernetes.NewForConfig(c)
```
Comment (mrogers950) [inline pkg/config/secrets.go:25]: nit: could be NewForConfigOrDie() and save a few code lines

File: pkg/config/secrets.go [hunk 2]
```diff
@@ -1,16 +1,41 @@
...
+		log.Fatalf("Failed to build ClientSet for Kubernetes cluster: %s", err)
+	}
+
+	n := v.GetString("database.password.secret_name")
+	ns := v.GetString("database.password.secret_namespace")
+	sm, err := cs.CoreV1().Secrets(ns).Get(context.TODO(), n, metav1.GetOptions{})
+	if err != nil {
+		log.Fatalf("Failed to retrieve secret: %s", err)
+	}
+	return string(sm.Data["password"])
```
Comment (mrogers950) [inline pkg/config/secrets.go:36]: Is "password" the required key that's defined elsewhere? (wondering if it would ever be something else or get changed)

Comment (rhmdnd) [reply pkg/config/secrets.go:36]: Yeah - technically it's a part of the config map. If someone really wanted to they could change this key.

**Comment by rhmdnd** [comment]
/retest

Not sure why exactly this failed since we're not adding anything for a deployment perspective.

**Pull request merged**